xbmc/peripherals/devices/Peripheral.cpp

```diff
@@ -149,10 +150,21 @@ bool CPeripheral::Initialise(void)
     return bReturn;
 
   g_peripherals.GetSettingsFromMapping(*this);
-  m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s.xml",
-                                          PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
-                                          m_strVendorId.c_str(),
-                                          m_strProductId.c_str());
+  if (m_iVendorId == 0x0000 && m_iProductId == 0x0000)
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
+  }
+  else
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            m_strVendorId.c_str(),
+                                            m_strProductId.c_str(),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
```

Comment:
I was thinking of changing the second block: 

else
 {
// Backwards compatibility 
 m_strSettingsFile = (without name) 
if! CFile::Exists( m_strSettingsFile) 
 m_strSettingsFile = (with name) 
} 

I prefer to replace spaces (and non-url-safe characters) with underscores. that's what I do in peripheral.joystick. see https://github.com/kodi-game/peripheral.joystick/blob/master/src/storage/StorageUtils.cpp#L58